index.js

```diff
@@ -67,28 +67,46 @@ var Node = function (path, opts, disposable) {
         })
       }
     },
-    daemon: function (cb) {
+    startDaemon: function (cb) {
       var t = this
       parseConfig(t.path, function (err, conf) {
-        var running = run(IPFS_EXEC, ['daemon'], {env: t.env})
-        t.pid = running.pid
-        running
+        if (err) return cb(err)
+
+        t.subprocess = run(IPFS_EXEC, ['daemon'], {env: t.env})
           .on('error', function (err) {
             if ((err + '').match('daemon is running')) {
               // we're good
               cb(null, ipfs(conf.Addresses.API))
+            } else if ((err + '').match('non-zero exit code')) {
+              // ignore when kill -9'd
             } else {
               cb(err)
             }
           })
           .on('data', function (data) {
-            var match = (data + '').trim().match(/API server listening on/)
+            var match = (data + '').trim().match(/API server listening on (.*)/)
             if (match) {
-              cb(null, ipfs(conf.Addresses.API))
+              t.apiAddr = match[1]
+              cb(null, ipfs(t.apiAddr))
             }
           })
       })
     },
+    stopDaemon: function () {
+      var t = this
+      var sp = t.subprocess
+
+      if (sp) {
+        sp.kill('SIGHUP')
+        setTimeout(function () {
+          sp.kill('SIGKILL')
```

Comment:
The problem with this is that i don't think there's a platform-independant way of doing this. For now i moved this into the tests, which we can run on platforms supporting 'kill' pid. No idea on how to do this in windows though.

I might be missing something though. But the sp.kill() here for example, returns immediatly, and you kind of don't know if it worked, just if it could find the pid or not.